snippets/curl/complete_migration.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/sh
+
+#####################n#####################################################
+
+# This script demonstrates how to complete the CA migration.
+
+# This script requires jq command-line JSON parser
+# if your system does not have jq installed, this will not work.
+# jq can be downloaded from here: https://github.com/stedolan/jq/releases
+
+###########################################################################
+
+port=1556
+master_server=""
+login_username=""
+login_password=""
+login_domainname=""
+login_domaintype=""
+reason=""
+force=0
+
+showHelp()
+{
+	echo ""
+	echo "Invalid command parameters"
+	echo "Usage:"
+	echo "./complete_migration.sh -nbmaster <master_server> -login_username <login_username> -login_password <login_password> -login_domainname <login_domain_name> -login_domaintype <login_domaintype> [-reason | -r <reason_for_migration>] [-force | -f]"
+	echo "-nbmaster         : Name of the NetBackup master server"
+	echo "-login_username   : User name of the user performing action"
+	echo "-login_password   : Password of the user performing action"
+	echo "-login_domainname : Domain name of the user performing action"
+	echo "-login_domaintype : Domain type of the user performing action"
+	echo "-reason | -r      : Reason for completing the CA migration"
+	echo "-force | -f       : Forcefully complete the CA migration"
+	echo ""
+	exit 1
+}
+
+parseArguments()
+{
+	if [ $# -ne 10 ] && [ $# -ne 11 ] && [ $# -ne 12 ] && [ $# -ne 13 ]; then
+		showHelp
+	fi
+
+	while [ "$1" != "" ]; do
+		case $1 in
+			-nbmaster)
+				master_server=$2
+				;;
+			-login_username)
+				login_username=$2
+				;;
+			-login_password)
+				login_password=$2
+				;;
+			-login_domainname)
+				login_domainname=$2
+				;;
+			-login_domaintype)
+				login_domaintype=$2
+				;;
+			-force|-f)
+				force=1
```

Comment:
Since `-force` doesn't consume `$2`, it _must_ be the final parameter on the command line. I realize it's also listed last in the synopsis, but that's a _really_ subtle detail. Most programs will accept options in _any_ order, and indeed, this program accepts all the _other_ options in any order.

You can fix this by using `shift` in each case branch according to how many arguments each branch consumes, instead of assuming each branch always consumes two arguments.